### PR TITLE
Add download report button  in dependent immunization tab for AB#12863

### PR DESCRIPTION
**Pull request by BrianMaki**
# Fixes or Implements [AB#12863](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/12863), [AB#13003](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/13003),  [AB#13009](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/13009), [AB#13006](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/13006)

## Description

1. Implemented download button for Dependent Immunization - history/forecast tabs.
2. Updated dependent immunization table on column - added Product - see [AB#13006](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/13006). 
3. Added loading component when downloading both immunization and COVID 19 reports.
4. Refactored DependentCardConponent in terms of function ordering as well as some code clean up.
5. Added report templates for server side code.
6. Updated Gateway API project with these changes as well. 

**Download Button Options:**

Screen Shot 2022-05-04 at 2 38 57 PM

**Immunization history tab with no data:**

Screen Shot 2022-05-04 at 2 38 38 PM

**Immunization history tab with data:**

Screen Shot 2022-05-04 at 2 38 13 PM

**Immunization forecast tab with no data:**

Screen Shot 2022-05-04 at 2 38 29 PM

**Immunization forecast tab with data:**

Screen Shot 2022-05-04 at 2 38 22 PM

**PDF report:**

[HealthGatewayDependentImmunizationReport.pdf](https://github.com/bcgov/healthgateway/files/8627671/HealthGatewayDependentImmunizationReport.pdf)

**CSV Report:**

[HealthGatewayDependentImmunizationReport.csv](https://github.com/bcgov/healthgateway/files/8627672/HealthGatewayDependentImmunizationReport.csv)

**XLSX Report:**

[HealthGatewayDependentImmunizationReport.xlsx](https://github.com/bcgov/healthgateway/files/8627669/HealthGatewayDependentImmunizationReport.xlsx)

If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

## Testing

-   [x] Unit Tests Updated
-   [ ] Functional Tests Updated
-   [ ] Not Required

### UI Changes

YES

### Browsers Tested

-   [x] Chrome
-   [ ] Safari
-   [ ] Edge
-   [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant. Include any specialized deployment steps here.

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

-   Fulfills Work Item Requirements

    -   The changes meet/implement story's acceptance criteria. Fixes identified problems if bug.
    -   Changes not directly related to the task requirements need to be documented on the PR.

-   Compilation / Tests

    -   The changes do not break compilation and/or unit or functional tests; unless other item addresses them specifically.

-   Logic Problems / Functional Behaviour

    -   The changes work as intended and do not introduce problems.

-   Performance

    -   The changes do not introduce obvious performance issues.

-   Documented Standards

    -   The changes adhere to the team's documented [Coding Standards](https://github.com/bcgov/healthgateway/wiki/standards).

-   Readability / Maintainability
    -   The changes can be easily understood/read and allow for future enhancements without major refactoring. Readability is preferred over "clever code".
    -   Reasoning for changes needs to be clearly articulated. Disagreements should be arbitrated by a third developer.

### CODE REVIEW

**Review by sslaws — APPROVED** [review]

File: Apps/WebClient/src/ClientApp/src/components/DependentCardComponent.vue [hunk 1]
```diff
@@ -109,10 +123,43 @@ export default class DependentCardComponent extends Vue {
...
+            dateOfBirth: this.formatDate(
+                this.dependent.dependentInformation.dateOfBirth || ""
+            ),
+            name: this.dependent.dependentInformation
+                ? this.dependent.dependentInformation.firstname +
+                  " " +
+                  this.dependent.dependentInformation.lastname
+                : "",
+            isRedacted: false,
+            datePrinted: new DateWrapper(new DateWrapper().toISO()).format(),
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/components/DependentCardComponent.vue:146]: Is the `.toISO()` needed or can this line be simplified? e.g.:
```
datePrinted: new DateWrapper().format(),
```

File: Apps/WebClient/src/ClientApp/src/components/DependentCardComponent.vue [hunk 2]
```diff
@@ -387,98 +589,42 @@ export default class DependentCardComponent extends Vue {
...
-            })
-            .catch((err: ResultError) => {
-                this.logger.error(err.resultMessage);
-                this.addError({
-                    errorType: ErrorType.Download,
-                    source: ErrorSourceType.Covid19LaboratoryReport,
-                    traceId: err.traceId,
-                });
-            });
+    private showCovid19DownloadConfirmationModalModal(
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/components/DependentCardComponent.vue:592]: Typo: ModalModal should just be Modal.

**Review by MikeLyttle — APPROVED** [review]

File: Apps/WebClient/src/ClientApp/src/components/DependentCardComponent.vue (hunk 1 above)
Comment (BrianMaki) [reply Apps/WebClient/src/ClientApp/src/components/DependentCardComponent.vue]: Changed.

File: Apps/WebClient/src/ClientApp/src/components/DependentCardComponent.vue (hunk 2 above)
Comment (BrianMaki) [reply Apps/WebClient/src/ClientApp/src/components/DependentCardComponent.vue]: Changed.

**Review by sslaws — APPROVED** [review]

**Pull request merged**